### PR TITLE
Add symlink and libs artifact types

### PR DESCRIPTION
**Pull request by cpuguy83**
Adds support for library files and package-level symlinks.
These are required for supporting packages such as golang.

Also adds more in-depth docs for artifacts.

### CODE REVIEW

File: website/docs/artifacts.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+## Artifact Configuration
+
+Most artifact types share a common data type so can be configured similarly.
+It is shown here as a reference which is linked to in the artifact descriptions
+where it is pertinent.
+
+Configuration options:
+
+- *subpath*(string): The provided path is joined to the typicall install path,
```
Comment (adamperlin) [inline website/docs/artifacts.md:19]:
```suggestion
- *subpath*(string): The provided path is joined to the typical install path,
```

File: website/docs/artifacts.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                     installed to.
+- *mode*(octal): file permissions to apply to the artifact.
+
+
+### Binaries
+
+Binaries are binary files that may be executed.
+On Linux these would typically get installed into `/usr/bin`.
+
+Binaries are a mapping of file path to [artifact configuration](#artifact-configuration).
```
Comment (adamperlin) [inline website/docs/artifacts.md:30]:
```suggestion
Binaries is a mapping of file path to [artifact configuration](#artifact-configuration).
```

File: website/docs/artifacts.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+The file path is the path to a file that must be available after the build
+section has finished.
```
Comment (adamperlin) [inline website/docs/artifacts.md:32]:
```suggestion
section has finished. This path is relative to the directory containing the listed sources in the spec. 
```

File: website/docs/artifacts.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+Example:
+
+```yaml
+artifacts:
+  binaries:
+    build_output/my_bin:
```
Comment (adamperlin) [inline website/docs/artifacts.md:39]:
```suggestion
    my_src/my_bin:
```
Can you change this to make it clear that generally, output binaries will be extracted from underneath source dirs?

File: website/docs/artifacts.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+```
+
+You may use a trailing wildcard to specify multiple binaries in a directory,
+though behavior may differ between different OS's/distros.
+
+### Links
+
+Links are a list of symlinks to be included with the package.
+Unlike most other artifact typtes, links do not reference any specific build
+artifact but rather a literal source to target mapping for the symlink.
```
Comment (adamperlin) [inline website/docs/artifacts.md:210]:
```suggestion
artifact but rather a literal source-to-target mapping for the symlink.
```

File: website/docs/artifacts.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+        enable: false
+        name: ""
+```
+
+### Libs
+
+Libs are library files to be included with your package.
+On Linux these typically go under `/usr/lib/<package>`.
+
+Licenses are a mapping of file path to [artifact configuration](#artifact-configuration).
```
Comment (adamperlin) [inline website/docs/artifacts.md:191]: Is this a duplication of the licenses section above?

File: website/docs/artifacts.md [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+What location this is depends on the target OS/distro and the kind of artifact.
+
+
+## Artifact Configuration
+
+Most artifact types share a common data type so can be configured similarly.
+It is shown here as a reference which is linked to in the artifact descriptions
+where it is pertinent.
+
+Configuration options:
```
Comment (adamperlin) [inline website/docs/artifacts.md:17]:
```suggestion
Configuration options shared by most artifacts:
```

File: website/docs/artifacts.md [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+      mode: 0o755
+```
+
+You may use a trailing wildcard to specify multiple binaries in a directory,
+though behavior may differ between different OS's/distros.
+
+### Data Dirs
+
+Data dirs are a list of read-only, architecture-independent data files.
+On Linux these typically get placed into `/usr/share` on Linux.
```
Comment (adamperlin) [inline website/docs/artifacts.md:70]:
```suggestion
On Linux these typically get placed into `/usr/share`.
```

File: website/docs/artifacts.md [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+
+### Directories
+
+Directories allows you to create new directories when installing the package.
+Two types of directory artifacts are supported:
+
+1. *config*: This is a directory where configuration files typically go, e.g. /etc/my_package2. *State*: This is directory for persistant state, typically in `/var/lib` on Linux.
+
+
+Unlinke many other artifact types, this does not reference any file produced
```
Comment (adamperlin) [inline website/docs/artifacts.md:92]:
```suggestion
Unlike many other artifact types, this does not reference any file produced
```

**Comment by cpuguy83** [comment]
Updated with suggestions.

**Review by adamperlin — APPROVED** [review]
I think this looks good to me now!

**Pull request merged**